ompi/mca/coll/base/coll_base_alltoallv.c

```diff
@@ -217,32 +227,72 @@ ompi_coll_base_alltoallv_intra_pairwise(const void *sbuf, const int *scounts, co
     ompi_datatype_type_size(sdtype, &sdtype_size);
     ompi_datatype_type_size(rdtype, &rdtype_size);
 
-    if (0 == sdtype_size || 0 == rdtype_size) {
+    if ((0 == sdtype_size || 0 == rdtype_size) && !SEND_0BYTES_ANYWAYS) {
```

Comment:
While we make this change in alltoallv, I think we should also refactor ialltoallv.

Specifically, my previous 2 changes should be revisited
- https://github.com/open-mpi/ompi/commit/a3a478750b64cbb8368a4fbfbe94eea57b9d996c
- https://github.com/open-mpi/ompi/commit/a7b4d3ebaea7ba2d1825d662de789722416c6ce1

With the solution in this PR, I suggest:
1. Remove `SEND_0BYTES_ANYWAYS`
2. Remove `0 == sdtype_size || 0 == rdtype_size` check - each send and recv should be based on `0 < count && 0 < dtype_size`. I realize my previous change was also wrong in that `sdtype_size` and `rdtype_size` should be evaluated separately.